gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -429,15 +596,28 @@ def _compute_metrics(
     return GtsfmMetricsGroup("translation_averaging_metrics", ta_metrics)
 
 
-def cast_to_measurements_variable_in_global_coordinate_frame(
-    i2Ui1_dict: Dict[Tuple[int, int], Optional[Unit3]], wRi_list: List[Optional[Rot3]], noise_model: gtsam.noiseModel
-) -> BinaryMeasurementsUnit3:
+def get_valid_measurements_in_world_frame(
+    i2Ui1_dict: Dict[Tuple[int, int], Optional[Unit3]], wRi_list: List[Optional[Rot3]]
+) -> Tuple[RelativeDirectionsDict, Set[int]]:
+    """Returns measurements for which both cameras have valid rotations, transformed to the world frame.
+
+    Args:
+        i2Ui1_dict: Relative translation directions between camera pairs.
+        wRi_list: List of estimated camera rotations.
 
-    w_i2Ui1_measurements = BinaryMeasurementsUnit3()
+    Returns:
+        Tuple of:
+            Relative translation directions between camera pairs, in world frame.
+            Set of camera indices for which we have valid rotations and measurements.
+    """
+
+    w_i2Ui1_dict = {}
+    valid_cameras: Set[int] = set()
     for (i1, i2), i2Ui1 in i2Ui1_dict.items():
         wRi2 = wRi_list[i2]
         if i2Ui1 is not None and wRi2 is not None:
-            # TODO: what if wRi2 is None, but wRi1 is not? Can we still transform.
```

Comment:
Why are we enforcing both cameras to have a global rotation. It makes sense to do it, but why do it now?